### PR TITLE
Hide WPCOM coming soon page when lys feature flag is enabled

### PR DESCRIPTION
**Pull request by chihsuan**
### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/wc-calypso-bridge/issues/1498.

Use the filter [a8c_show_coming_soon_page](https://github.com/Automattic/jetpack/blob/065c5cbec110192aeef1bfb37124a2351251a00e/projects/packages/jetpack-mu-wpcom/src/features/coming-soon/coming-soon.php#L40C25-L40C50) to override WPCOM's coming soon status so that we can show LYS coming soon page instead.

There a another task to [sync WPCOM coming soon status to LYS](https://github.com/woocommerce/team-ghidorah/issues/344). When that is done, we should see WooCommerce's coming soon template shown, when in coming soon mode either on LYS or WPCOM.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Use a WPCOM atomic e-commerce site and local wc-calypso-bridge
2. Enable LYS feature flag (via beta tester or you can remove [these lines](https://github.com/Automattic/wc-calypso-bridge/blob/507bde7b933c60bb2cb7efdfad66b7bd8581b39b/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L143-L147).
3. Go to `https://wordpress.com/settings/general/SITE_URL` and enable WPCOM coming soon
4. Go to `/wp-admin/admin.php?page=wc-settings&tab=site-visibility` and enable LYS coming soon
5. Open the site frontend in an incognito window and confirm that the LYS coming soon page is shown

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/class-wc-calypso-bridge-coming-soon.php
Comment (ilyasfoo) [inline includes/class-wc-calypso-bridge-coming-soon.php:40]: I think typically we would use this if we intend to run it last

```suggestion
		add_filter( 'a8c_show_coming_soon_page', array( $this, 'should_show_a8c_coming_soon_page' ), PHP_INT_MAX, 1 );
```

**Review by ilyasfoo — APPROVED** [review]
Tests well, thanks @chihsuan! We'll also need to update the version in `class-wc-calypso-bridge.php` to `x.x.x`.

I have minor comment but LGTM!

File: includes/class-wc-calypso-bridge-coming-soon.php
Comment (ilyasfoo) [inline includes/class-wc-calypso-bridge-coming-soon.php:50]: I forgot, there was some instances where `Features` class was not available during a run. I'm unsure how early `should_show_a8c_coming_soon_page` is run, but the prior art is to check for the class too: https://github.com/Automattic/wc-calypso-bridge/blob/3233649621c32b4f7f18daf68ee381f90a7df970/includes/class-wc-calypso-bridge-ecommerce-admin-menu.php#L380

What do you think?

**Review by moon0326 — COMMENTED** [review]
LGTM and tested well 👍 🚀 

I'll let @ilyasfoo to approve it.

File: includes/class-wc-calypso-bridge-coming-soon.php
Comment (chihsuan) [reply includes/class-wc-calypso-bridge-coming-soon.php]: That's a good idea. Updated in  3038da3be2d72c9b5866676a94289233cbb528bc

Comment (chihsuan) [reply includes/class-wc-calypso-bridge-coming-soon.php]: I wonder if there is anything we can do to prevent any future issues like this.  🤔  We also use `Features::is_enabled` directly here, though it should be safe to use it as it is in the setup task context. In other places, we use methods/classes from WooCommerce core without checking their availability. 

https://github.com/Automattic/wc-calypso-bridge/blob/717d11285b7b770ee7dbbc6e8251d7646ade4dfc/includes/class-wc-calypso-bridge-setup-tasks.php#L147-L153

  I'd also like to know why the Features class was unavailable during the run.

**Review by ilyasfoo — APPROVED** [review]
Thanks @chihsuan! The change tests well.

File: includes/class-wc-calypso-bridge-coming-soon.php
Comment (ilyasfoo) [reply includes/class-wc-calypso-bridge-coming-soon.php]: Good question, @chihsuan! I think it's one of the pitfalls of WordPress' filters. I believe in some cases the filter ran before WooCommerce initialized.

I think a good solution might be to create a class helper that runs a class callback after WooCommerce has initialized, but I'm not sure how that'll look

Comment (ilyasfoo) [reply includes/class-wc-calypso-bridge-coming-soon.php]: To be accurate, it's very much possible that the `should_show_a8c_coming_soon_page` filter is safe. I was just trying to be guarantee a safe execution, since it's possible for other mu-plugins to do a `apply_filter( 'should_show_a8c_coming_soon_page' )` before WooCommerce is loaded

Comment (chihsuan) [reply includes/class-wc-calypso-bridge-coming-soon.php]: Good point! Agreed that it's better to be safe.

**Pull request merged**
